### PR TITLE
Document encrypted backup process

### PR DESCRIPTION
**Pull request by jdbohrman**
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/5109)

 
 
 

Docs Preview: [Dgraph Preview](https://dgraph-d7073c833e-55985.surge.sh)

### CODE REVIEW

File: wiki/content/enterprise-features/index.md [hunk 1]
```diff
@@ -703,3 +702,47 @@ Here's an example to run bulk loader with a key used to write encrypted data:
 ```bash
 dgraph bulk --encryption_key_file "./enc_key_file" -f data.json.gz -s data.schema --map_shards=1 --reduce_shards=1 --http localhost:8000 --zero=localhost:5080
 ```
+
+## Encrypted Backups
```
Comment (parasssh) [inline wiki/content/enterprise-features/index.md:706]: I think we dont need a separate section for Encrypted Backups. The documentation for this feature can be inserted into the binary backups topic elsewhere in this doc.

File: wiki/content/enterprise-features/index.md [hunk 2]
```diff
@@ -703,3 +702,47 @@ Here's an example to run bulk loader with a key used to write encrypted data:
...
+
+Encrypted backups are a Enterprise feature that allow you to encrypt your backups and restore them. This documentation describes how to implement encryption into your binary backups
+
+### New flag “Encrypted” in manifest.json
```
Comment (parasssh) [inline wiki/content/enterprise-features/index.md:710]: This section is an implementation detail. I dont think we need to document this.

File: wiki/content/enterprise-features/index.md [hunk 3]
```diff
@@ -703,3 +702,47 @@ Here's an example to run bulk loader with a key used to write encrypted data:
...
+
+A new flag “Encrypted” is added to the `manifest.json`. This flag indicates if the corresponding binary backup is encrypted or not. To be backward compatible, if this flag is absent, it is presumed that the corresponding backup is not encrypted.
+
+For a series of full and incremental backups, per the current design, we don't allow mixing of encrypted and unencrypted backups. As a result, all the full and incremental backups must be all encrypted or not. This flag helps with checking this restriction.
+
+NOTE: If a mismatch is found (e.g. latest backup is encrypted but current alpha is not configured with encryption or vice-versa), an error is thrown. The user must then force a full backup using the `forcefull` flag in the backup API to force a backup.
+
+### AES And Chaining with Gzip
```
Comment (parasssh) [inline wiki/content/enterprise-features/index.md:718]: This section is an implementation detail. I dont think we need to document this.

File: wiki/content/enterprise-features/index.md [hunk 4]
```diff
@@ -703,3 +702,47 @@ Here's an example to run bulk loader with a key used to write encrypted data:
...
+
+If encryption is turned on an alpha, then we use the configured encryption key. The key size (16, 24, 32 bytes) determines AES-128/192/256 cipher chosen. We use the AES CTR mode. Currently, the binary backup is already gzipped. With encryption, we will encrypt the gzipped data. 
+
+### Initialization Vector
```
Comment (parasssh) [inline wiki/content/enterprise-features/index.md:722]: This section is an implementation detail. I dont think we need to document this.

File: wiki/content/enterprise-features/index.md [hunk 5]
```diff
@@ -703,3 +702,47 @@ Here's an example to run bulk loader with a key used to write encrypted data:
...
+
+The AES Cipher should use a random Initialization Vector (IV). This is a random sequence of 16 bytes. It must be unique but not a secret to obtain desirable security properties (See https://en.wikipedia.org/wiki/Initialization_vector ) . As a convention, the IV is inserted alongside the cipher text.
+
+During **backup**: the 16 bytes IV is prepended to the Cipher-text data after encryption.
+
+During **restore**: the first 16 bytes is read as the IV before decrypting the rest of the data.
+
+### Backup
```
Comment (parasssh) [inline wiki/content/enterprise-features/index.md:730]: Merge this and the next section with the backups section. You can add some bits of information from the implementation details but dont need to add whole sections on those.

File: wiki/content/enterprise-features/index.md [hunk 6]
```diff
@@ -703,3 +702,47 @@ Here's an example to run bulk loader with a key used to write encrypted data:
...
+
+Backup is an online tool, meaning it is available when alpha is running. For encrypted backups, the alpha must be configured with the “encryption-key-file”. 
+
+{{% notice "note" %}}
+encryption-key-file was used for encryption-at-rest and will now also be used for encrypted backups.
```
Comment (parasssh) [inline wiki/content/enterprise-features/index.md:735]: encryption_key_file

File: wiki/content/enterprise-features/index.md [hunk 7]
```diff
@@ -703,3 +702,47 @@ Here's an example to run bulk loader with a key used to write encrypted data:
 ```bash
 dgraph bulk --encryption_key_file "./enc_key_file" -f data.json.gz -s data.schema --map_shards=1 --reduce_shards=1 --http localhost:8000 --zero=localhost:5080
 ```
+
+## Encrypted Backups
+
+Encrypted backups are a Enterprise feature that allow you to encrypt your backups and restore them. This documentation describes how to implement encryption into your binary backups
```
Comment (sleto-it) [inline wiki/content/enterprise-features/index.md:708]: Please let's add from when they are available: v20.03.1 and v1.2.3

**Comment by parasssh** [comment]
@jdbohrman  please cherry-pick this release/v20.03 and release/v1.2

**Pull request merged**